.kokoro/deps-finder.sh

```diff
@@ -0,0 +1,202 @@
+
+#Returns 1 if we find an updated dependency in the commit message.
+function getDependency() {
+  currCommitNumber=$(git rev-parse HEAD)
+
+  #Note that this grep serves to check if we're updating a dependency including shared-dependencies.
+  currCommitMessage=$(git log -1 $currCommitNumber | grep "deps: update dependency com.google.cloud:google-cloud-")
+
+  if [[ -z $currCommitMessage ]]
+  then
+    echo "Commit message does not have correct formatting. Assuming no error."
+    return 1
+  fi
+
+  dependency=${currCommitMessage:28}
+  dependency=${dependency/ to v/:}
+  dependency=${dependency/*com.google.cloud:}
+
+  if [[ -z $dependency ]]
+  then
+    return 1
+  fi
+
+  echo "$dependency"
+  return 0
+}
+
+function getLatestSharedDepsVersion() {
+  group="com.google.cloud"
+  artifact="google-cloud-shared-dependencies"
+  latest="$group:$artifact:LATEST"
+  pom=$(mvn help:effective-pom -Dartifact=$latest)
+
+  if [[ -z $pom ]]
+  then
+    echo "ERROR: Failed to find effective POM!"
+    return 1
+  fi
+
+  sharedDepsLatestVersion="$(echo "$pom" | grep "$group:$artifact")"
+  sharedDepsLatestVersion=${sharedDepsLatestVersion/* \'/}
+  sharedDepsLatestVersion=${sharedDepsLatestVersion/\' */}
+  sharedDepsLatestVersion=${sharedDepsLatestVersion/*:}
+  if [[ -z $sharedDepsLatestVersion ]]
+  then
+    echo "ERROR: Failed to find latest shared config version!"
+    return 1
+  fi
+
+  echo "$sharedDepsLatestVersion"
+  return 0
+}
+
+#Pass in dependency to script.
+#Dependency should be of the form artifactId:version
+
+function checkDependency() {
+  dependency=$1
+
+  #If no dependency was passed straight to the script, check our most recent commit for a dep version update.
+  if [[ -z $dependency ]]
+  then
+    dependency=$(getDependency)
+  else
+    if [[ -z $(echo $dependency | grep google-cloud-) ]]
+    then
+      echo "$dependency does not require shared-dependencies criteria. Returning success."
+      return 0
+    fi
+  fi
+
+  #We did not successfully find a valid commit with a dependency. Return success - nothing to check here!
+  if ! [[ $? -eq 0 ]]
+  then
+    echo "Commit message does not have correct formatting. Assuming no error."
+    return 0
+  fi
+
+  #Should just be the desired version for shared-deps, e.g. "0.8.3"
+  desiredDeps=$2
+
+  if [[ -z $desiredDeps ]]
+  then
+    desiredDeps=$(getLatestSharedDepsVersion)
+    if [[ -z $desiredDeps ]]
+    then
+      echo "Unable to find correct version of shared-dependencies."
+      return 1
+    fi
+  fi
+
+  depVersion=${dependency/*:} #Grab last part of dependency artifactId:version input.
+  depVersion=${depVersion/-*} #Remove any alphabetical characters at the end (e.g. -alpha, -beta, etc)
+
+  pomLocation="/pom.xml"
+
+  #The naming of these libraries does not follow the rules. As such, we've added in special cases for them.
+  #depAID=dependency artifact ID - this is the ID inside of the Github link for the dependency.
+  case $dependency in
```

Comment:
For artifact -> repository there should be a `scm` field in the artifact's pom.xml

Reply:
Most of the libraries follow naming conventions that make them reasonably easy to find - it's currently only these couple that have issues. Were we to try the scm field, we would likely need to grab the effective POM from Maven, grab the link from the Github repository from there, manipulate the string, then connect to the POM. In terms of logic, this seemed to be simpler